File: sdks/go/pkg/models/shared/recentitems.go
package shared

// RecentItems
// The part of an API response that describes marker
// based pagination
type RecentItems struct {
	Entries    []RecentItem `json:"entries,omitempty"`
	Limit      *int64       `json:"limit,omitempty"`
	NextMarker *int64       `json:"next_marker,omitempty"`
	PrevMarker *int64       `json:"prev_marker,omitempty"`
}

File: sdks/ts/src/sdk/models/shared/recentitems.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { RecentItem } from "./recentitem";



// RecentItems
/** 
 * The part of an API response that describes marker
 * based pagination
**/
export class RecentItems extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=entries", elemType: RecentItem })
  entries?: RecentItem[];

  @SpeakeasyMetadata({ data: "json, name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "json, name=next_marker" })
  nextMarker?: number;

  @SpeakeasyMetadata({ data: "json, name=prev_marker" })
  prevMarker?: number;
}
